packages/notifications/push/amplify_push_notifications_pinpoint/lib/src/pinpoint_provider.dart

```diff
@@ -1,25 +1,221 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+import 'dart:convert';
+
 import 'package:amplify_core/amplify_core.dart';
 
-// TODO(Samaritan1011001): Complete the implementation using Analytics Client
-class PinpointProvider extends ServiceProviderClient {
+final AmplifyLogger _logger = AmplifyLogger.category(Category.pushNotifications)
+    .createChild('AmplifyPushNotification');
+
+/// {@template amplify_push_notifications_pinpoint.pinpoint_provider}
+/// AWS Pinpoint provider that implements [ServiceProviderClient].
+///
+/// [init] method has to be called before other methods can be used.
+/// Once initialized, it can [registerDevice], [recordNotificationEvent]
+/// & [identifyUser] with Pinpoint.
+/// {@endtemplate}
+class PinpointProvider implements ServiceProviderClient {
+  /// {@macro amplify_push_notifications_pinpoint.pinpoint_provider}
+  PinpointProvider() {
+    _androidCampaignIdKey += _campaginIdKey;
+    _androidCampaignActivityIdKey += _campaignActivityIdKey;
+    _androidCampaignTreatmentIdKey += _treatmentIdKey;
+  }
+
+  // TODO(Samaritan1011001): Enable use of FlutterAnalyticsClient after merging to feature branch
+  late dynamic _analyticsClient;
+  final _campaginIdKey = 'campaign_id';
+  final _campaignActivityIdKey = 'campaign_activity_id';
+  final _treatmentIdKey = 'treatment_id';
+  String _androidCampaignIdKey = 'pinpoint.campaign.';
+  String _androidCampaignActivityIdKey =
+      'pinpoint.campaign.campaign_activity_id';
+  String _androidCampaignTreatmentIdKey = 'pinpoint.campaign.treatment_id';
```

Comment:
Not sure what's going on here - everything but `_analyticsClient` should be `static const` it seems and should not change ever.

```suggestion
  PinpointProvider() {
    _androidCampaignIdKey += _campaginIdKey;
    _androidCampaignActivityIdKey += _campaignActivityIdKey;
    _androidCampaignTreatmentIdKey += _treatmentIdKey;
  }

  // TODO(Samaritan1011001): Enable use of FlutterAnalyticsClient after merging to feature branch
  late dynamic _analyticsClient;
  final _campaginIdKey = 'campaign_id';
  final _campaignActivityIdKey = 'campaign_activity_id';
  final _treatmentIdKey = 'treatment_id';
  String _androidCampaignIdKey = 'pinpoint.campaign.';
  String _androidCampaignActivityIdKey =
      'pinpoint.campaign.campaign_activity_id';
  String _androidCampaignTreatmentIdKey = 'pinpoint.campaign.treatment_id';
```

Reply:
Cleaned it up.